kylin-spark-project/kylin-spark-engine/src/main/scala/org/apache/kylin/engine/spark/job/CubeBuildJob.java

```diff
@@ -120,7 +120,7 @@ protected void doExecute() throws Exception {
                 infos.recordSpanningTree(segId, spanningTree);
 
                 logger.info("Updating segment info");
-                updateSegmentInfo(getParam(MetadataConstants.P_CUBE_ID), seg, buildFromFlatTable.getCount());
+                updateSegmentInfo(getParam(MetadataConstants.P_CUBE_ID), seg, buildFromFlatTable.getParentDS().count());
```

Comment:
What's the reason of `getParentDS`?

Reply:
Well, it'd be better to use getFlattableDS.